pytest_html/plugin.py

```diff
@@ -135,6 +135,26 @@ def __lt__(self, other):
                      'XPassed', 'Skipped', 'Passed')
             return order.index(self.outcome) < order.index(other.outcome)
 
+        def save_external_src(self, content, extra_index,
+                              test_index, file_extension):
+            hash_key = ''.join([self.test_id, str(extra_index),
+                               str(test_index)]).encode('utf-8')
+            hash_generator = hashlib.md5()
+            hash_generator.update(hash_key)
+            image_file_name = '{0}.{1}'.format(hash_generator.hexdigest(),
+                                               file_extension)
+            image_path = os.path.join(os.path.dirname(self.logfile),
+                                      'assets', image_file_name)
+            if not os.path.exists(os.path.dirname(image_path)):
+                os.makedirs(os.path.dirname(image_path))
+
+            relative_path = '{0}/{1}'.format('assets', image_file_name)
+
+            with open(image_path, 'w') as f:
+                f.write(content)
+            return relative_path
+
+
```

Comment:
@davehunt there are some flake8 problems and it is missing some tests, but I hope you can take a look before you go  =)